### PR TITLE
Embedded hal 1.0.0

### PR DESCRIPTION
**Pull request by sjoerdsimons**
Hi! Thank you for helping out with SSD1306 development! Please:

- [x] Check that you've added documentation to any new methods
- [x] Rebase from `master` if you're not already up to date
- [x] Add or modify an example if there are changes to the public API
- [x] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, **Changed**, etc)
- [x] Run `rustfmt` on the project with `cargo fmt --all` - CI will not pass without this step
- [x] Check that your branch is up to date with master and that CI is passing once the PR is opened

## PR description

This cherry-picks the embedded-hal-1.0.0 update from #207  and combined that with changing the various examples to use embassy-stm32 as the pac provider which implements embedded-hal-1.0 (stm32f1xx-hal does not yet).

### CODE REVIEW

**Comment by sjoerdsimons** [comment]
@eldruin any chance for a review :) Fwiw the minor version ofcourse needs to be bumped, unsure if you prefer that in the PR or to do it directly after merging

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ SSD1306 monochrome OLED display.
 
 ## [Unreleased] - ReleaseDate
 
+- Changed dependencies for embbedded-hal-1.0.0
+- Switch examples to embassy stm32 pac whith implemented embedded-hal-1.0.0
+
```
Comment (eldruin) [inline CHANGELOG.md:12]:
```suggestion
- **(breaking)** Increased MSRV to 1.75.0

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ SSD1306 monochrome OLED display.
 
 ## [Unreleased] - ReleaseDate
 
+- Changed dependencies for embbedded-hal-1.0.0
```
Comment (eldruin) [inline CHANGELOG.md:10]:
```suggestion
- Updated dependencies for `embedded-hal` 1.0.0.
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,9 +35,9 @@ jobs:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
         with:
-            toolchain: 1.61
-      - run: cargo test --lib --target x86_64-unknown-linux-gnu
-      - run: cargo test --doc --target x86_64-unknown-linux-gnu
+            toolchain: 1.75
+      - run: cargo build --lib --target x86_64-unknown-linux-gnu
+      - run: cargo doc --target x86_64-unknown-linux-gnu
```
Comment (eldruin) [inline .github/workflows/ci.yml:40]: Do the examples not build with Rust 1.75.0? If so, please rename this step "build with MSRV"

**Review by eldruin — COMMENTED** [review]
Looks great, thank you!

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,9 @@ SSD1306 monochrome OLED display.
...
+- Switch examples to embassy stm32 pac whith implemented embedded-hal-1.0.0
```
Comment (eldruin) [inline CHANGELOG.md:11]:
```suggestion
- Switch examples to embassy STM32 PAC which implements `embedded-hal` 1.0.0 traits.
```

File: .github/workflows/ci.yml (hunk 3 above)
Comment (sjoerdsimons) [reply .github/workflows/ci.yml:40]: Specifically embassy needs 1.79 at the moment; renamed now

**Comment by sjoerdsimons** [comment]
@eldruin oh one favour; if your merge this can you use a fast-forward/merge commit rather then squashing the lot so the seperated commits are maintained :)

**Review by eldruin — APPROVED** [review]
Looks good to me, thank you for your work!

**Pull request merged**